doc/dualtor/ICMP_Hardware_Offload_and_Protecion.md

```diff
@@ -0,0 +1,289 @@
+* **ICMP echo session rx_interval setting:**
+    * rx_interval = probing interval x negative signal count.
+    * LinkMgrd will directly transition the link prober to unknown state avoiding the negative probing timer as opposed to what was done in case of software probing.
+
+#### Link Prober State transitions
+Following table shows the link prober state transitions based on event when link_prober mode is set as hardware and icmp echo session is offloaded to NPU.
+<div align="center"> <img src=image/link_state_transition.png width=600 /> </div>
+
+#### TLV generation considerations
+ICMP packets with TLVs will not be generated by the hardware/NPU and LinkMgrd will generate these packets in software using the software cookie as currently done by software prober. Using the software cookie for TLV generation will make sure peer NPU will not consume these ICMP packets in hardware and LinkMgrd running on peer ToR will be able to receive and handle these TLV packets.
+
+### Orchagent
+#### IcmpOrch
+IcmpOrch is a new component introduced which consumes icmp echo session entries from App DB ICMP_ECHO_SESSION_TABLE and programs the ICMP hardware offload sessions in NPU. It receives icmp echo session state notifications from SAI / NPU and produces session state in State DB ICMP_ECHO_SESSION_TABLE that is consumed by LinkMgrd.
+
+Following diagram describes the component level flow for icmp session creation and removal.
+<div align="center"> <img src=image/icmp_session_create.png width=700 /> </div>
+
+Following diagram describes the component level flow for icmp state change notifications.
+<div align="center"> <img src=image/icmp_session_state_notification.png width=600 /> </div>
+
+* **Session Key format:**
+Vrf-name : Interface Alias : Session GUID : Session Type
+* Examples of session key format:
+  * default:default:5000:NORMAL
+  * default:default:6000:RX
+  * default:Ethernet0:12000:NORMAL
+* Session Type:
+  * 'NORMAL': Install a normal icmp echo session that will transmit, receive icmp echo session packets and monitor the session.
+  * 'RX': Install a monitor only icmp echo session and set the tx_interval to zero which will not send any icmp echo session.
+* Interface Alias 'default' will set the SAI atttribute hw_lookup to true and NPU will perform forwarding lookup basd on dst_ip to determine the outgoing interface.
+
+#### MuxOrch
+This feature introduces a new config knob **switching_mode** to differentiate between normal software based switching and FRR protection switching that will use next hop protection group to switch traffic. MuxOrch will create next hop protection group for each mux port and maintain a mapping of mux port and next hop protection group based on this config knob.

```

Comment:
As we have FRR (Free Range Routing) for routing stack, usage of FRR for Fast ReRoute is confusing, please use different name.